Update typescript.md to describe how to use typescript types without module bundler (closes #2042) (#2137)
vuejs/vuejs.org: src/v2/guide/typescript.md

}
```

> To use vue types in environment without module bundler you can create custom type definition file containing declare global (see https://www.typescriptlang.org/docs/handbook/declaration-merging.html)

Reviewer: ```suggestion
To use Vue types in an environment without a module bundler, you can create a custom type definition file containing `declared global` (see [Declaration Merging
](https://www.typescriptlang.org/docs/handbook/declaration-merging.html)).
```

> To use Vue types in an environment without a module bundler, you can create a custom type definition file containing `declare global` (see [Declaration Merging https://www.typescriptlang.org/docs/handbook/declaration-merging.html](https://www.typescriptlang.org/docs/handbook/declaration-merging.html)).

Reviewer 2: It would make more sense to explicitly describe what the definition file does.

> To use Vue types

I feel we cannot use Vue's types (like `ComponentOptions` etc.) without this definition which is not correct from this phrase.

What about the below change?

```suggestion
To use Vue in an environment without a module bundler, you can create a custom type definition file containing `declare global` to declare global `Vue` object type (see [Declaration Merging https://www.typescriptlang.org/docs/handbook/declaration-merging.html](https://www.typescriptlang.org/docs/handbook/declaration-merging.html)).
```

Author: Thank you all for reviewing. Yes, wording needs some work to make it clear what is the purpose of this. You can use Vue fine in typescript without importing types or creating custom type definition file with declare global if you can live with errors (javascript still gets generated correctly) and without intellisense and type recognition in IDE.

Author: ```suggestion
To compile Vue without errors and have intellisense and type recognition in IDE in environment without a module bundler you can create a custom type definition file containing `declare global` to declare global `Vue` object type (see [Declaration Merging https://www.typescriptlang.org/docs/handbook/declaration-merging.html](https://www.typescriptlang.org/docs/handbook/declaration-merging.html)).
```

Reviewer 2: Wait, is this code really work on your environment? I just try it in my local but there is a compile error on it. I'm using the latest Vue (v2.6.10).

```
Subsequent variable declarations must have the same type.  Variable 'Vue' must be of type 'typeof import("(...)/node_modules/vue/types/index")', but here has type 'VueConstructor<Vue>'.ts(2403)
```

Reviewer 2: OK, I realized there is already this kind of type declaration in Vue's typings. https://github.com/vuejs/vue/blob/dev/types/index.d.ts#L5

But this declaration is actually not correct as `Vue` object type is declared under `Vue.default`. This causes the compilation error which I showed above.

I already created the PR to fix this bug. Once this is merged, you can use global `Vue` without any additional set up which means we would no longer require this section in docs.
https://github.com/vuejs/vue/pull/9912

Author: Yes I think something has changed in latest typescript version that makes this kind of declaration break - I need to investigate more. But if we can make it work in global context without tricks then even better.

}
```

> To use Vue types in an environment without a module bundler, you can create a custom type definition file containing `declared global` (see [Declaration Merging

Reviewer: The markup is broken here.

Author: Not sure I understood exactly what is broken in markup? Found a typo and added link to url label as well.

Reviewer: Preview your markdown and you'll see what I mean.

Author: Interesting. I am probably missing something obvious because I did preview and looks correct to me. Here is what I am testing:

https://github.com/vuejs/vuejs.org/blob/6e6891eeb139c6dc712169359677e1bf72dd3c4b/src/v2/guide/typescript.md

And here is what I am seeing:

![image](https://user-images.githubusercontent.com/1261322/56531031-0104d000-655c-11e9-9bd8-b61f367fa504.png)
